server/vulnerabilities/msrc/io/msrc_test.go

```diff
@@ -32,7 +32,8 @@ func TestMSRCClient(t *testing.T) {
 			})
 
 			t.Run("provided month and year is in the future", func(t *testing.T) {
-				_, err := sut.GetFeed((now.AddDate(0, 1, 0)).Month(), now.Year())
+				future := now.AddDate(1, 1, 0)
```

Comment:
I don't think this is the proper fix. I understand this test as "any future date results in an error".

Reply:
I have a suspicion that this is failing due to the following code https://github.com/fleetdm/fleet/blob/add2f28305bd6196c4158c3497287217549b1d00/server/vulnerabilities/msrc/io/msrc.go#L60

```
	d := time.Date(year, month, 1, 0, 0, 0, 0, time.UTC) // this sets the hour to 1
	... 

	if d.After(time.Now().UTC()) { // but we are comparing with now ?! 
		return "", errors.New("date can't be in the future")
	}
```

I think this may be an issue on the last day of the month, and we add a month, and it's near 00:00 UTC...

Reply:
copying and pasting from a chat, for the sake of over-communication:

I'm certain that the test fails because it's December, the problematic line is:

```
				_, err := sut.GetFeed((now.AddDate(0, 1, 0)).Month(), now.Year())
```

`(now.AddDate(0, 1, 0)).Month()` returns `"January"` , and `now.Year()` returns `2022` , so we're actually sending a date in the past

Reply:
having said that, maybe we should do two or three assertions to make the intention of the test clearer? (future month, future year, future month + future year)

Reply:
Ah, you are right! I did not notice that. 

However, I also think there is a problem with the code in `GetFeed` as a pointed out above. I am not sure what the easiest way to test this would be, as it would probably require refactoring the code to use a mock clock/pass the current time as an argument.

Reply:
Adding a couple tests probably makes sense, especially because they are quite short.

Reply:
>  it would probably require refactoring the code to use a mock clock/pass the current time as an argument.

agreed! would opening an issue to investigate/fix work?

> Adding a couple tests probably makes sense, especially because they are quite short

sounds good! I added a couple more.

Reply:
> agreed! would opening an issue to investigate/fix work?

Yes, lets follow up. Thank you for merging this to fix CI quickly, but I want to make sure something else isn't wrong here either.

Reply:
sounds good, I created https://github.com/fleetdm/fleet/issues/8898 to follow up